Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM tuttleofx/tuttleofx-python3.5
+
+MAINTAINER ButtleOFX <buttleofx-dev@googlegroups.com>
+
+ENV BUTTLEOFX_DEV=/opt/ButtleOFX_git \
+    HOME=/home/developer
+
+RUN apt-get update && apt-get install -y \
+    python3 \
+    python3-pip \
+    qt5-default \
+    qtdeclarative5-dev \
+    python3-pyqt5 \
+    python3-pyqt5.qtquick \
+    qml-module-qtquick-controls \
+    qml-module-qtquick-layouts \
+    qml-module-qtquick-dialogs \
+    qml-module-qtquick-localstorage \
+    libgl1-mesa-dev \
+    mesa-utils \
+    && pip3 install numpy PyOpenGL PyOpenGL_accelerate \
+    && apt-get clean && apt-get autoremove && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# Replace 1000 with your user / group id
+RUN export uid=1000 gid=1000 && \
+    mkdir -p /home/developer && \
+    mkdir -p /etc/sudoers.d && \
+    echo "developer:x:${uid}:${gid}:Developer,,,:/home/developer:/bin/bash" >> /etc/passwd && \
+    echo "developer:x:${uid}:" >> /etc/group && \
+    echo "developer ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/developer && \
+    chmod 0440 /etc/sudoers.d/developer && \
+    chown ${uid}:${gid} -R /home/developer
+
+COPY . $BUTTLEOFX_DEV
+WORKDIR $BUTTLEOFX_DEV
+USER developer
```

Comment:
Yes we could, I though about it first but I think that developer is more
accurate if the user dives inside the image. In that case the user would
dev.

No preference , as you want.
Le 28 mai 2016 13:34, "Fabien Castan" <notifications@github.com> a écrit :

> In Dockerfile
> <https://github.com/buttleofx/ButtleOFX/pull/192#discussion_r64987849>:
>
> > +    && pip3 install numpy PyOpenGL PyOpenGL_accelerate \
> > +    && apt-get clean && apt-get autoremove && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
> > +
> > +# Replace 1000 with your user / group id
> > +RUN export uid=1000 gid=1000 && \
> > +    mkdir -p /home/developer && \
> > +    mkdir -p /etc/sudoers.d && \
> > +    echo "developer:x:${uid}:${gid}:Developer,,,:/home/developer:/bin/bash" >> /etc/passwd && \
> > +    echo "developer:x:${uid}:" >> /etc/group && \
> > +    echo "developer ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/developer && \
> > +    chmod 0440 /etc/sudoers.d/developer && \
> > +    chown ${uid}:${gid} -R /home/developer
> > +
> > +COPY . $BUTTLEOFX_DEV
> > +WORKDIR $BUTTLEOFX_DEV
> > +USER developer
>
> We could name the user buttle.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/buttleofx/ButtleOFX/pull/192/files/fa1bb32b0cc5fb377648a54f33204d3f88f3ed21..15116ba4f4be461d1f36acbaaffb47fe4a033076#r64987849>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG9JVoU5qqA0spGl6Krb_SB2YMylLX9Cks5qGCgugaJpZM4IouHh>
> .
>